src/components/Dashboard /PartnerData/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import "./styles.scss";
+// import { METHODS } from "../../../services/api";
+import { Link } from "react-router-dom";
+import { PATHS } from "../../../constant";
+import { useSelector } from "react-redux";
+
+function PartnerDashboard() {
+  const [partners, setPartners] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const user = useSelector(({ User }) => User);
+
+  useEffect(() => {
+    axios
+      .get(` https://api.merakilearn.org/partners`, {
+        headers: { Authorization: user.data.token },
+      })
+      .then((res) => {
+        setPartners(res.data);
+      });
+  }, []);
+
+  const handleSearchChange = (e) => {
+    e.preventDefault();
+    setSearchTerm(e.target.value);
+  };
+
+  // API CALL FOR GET partners DATA
+  // useEffect(() => {
+  //   axios({
+  //     method: METHODS.GET,
+  //     url: `${process.env.REACT_APP_MERAKI_URL}/partners`,
+  //     headers: {
+  //       accept: "application/json",
+  //       Authorization: user.data.token,
+  //     },
+  //   }).then((res) => {
+  //     setPartners(res.data);
+  //   });
+  // }, []);
+
+  return (
+    <>
+      <div className="table-search">
+        <input
+          type="text"
+          placeholder="Search..."
+          onChange={handleSearchChange}
```

Comment:
use a debounce function for `onChange`. Add this library and use it: https://github.com/xnimorz/use-debounce#readme

Reply:
I have doubt about this, when we need data from the API that time we use debounce. Here we are using the filter for searching data do we need to use debounce here?

Reply:
done